genbtf.sh

```diff
@@ -64,48 +97,75 @@ cat <<EOF
 
 EOF
 
-Commit=$(cd $1 && git rev-parse HEAD)
-if [ -z $Commit ]; then
-	exit 1
-fi
-
 printf "const char *btfhub_archive_commit=\"%s\";\n\n" "$Commit"
 
-# Table of all successfull kernels, so we can create all_btfs[]
-typeset -a Good
-typeset -i Total=0
+function do_arch
+{
 
-for s in $Srcs; do
-	distro=${s%/*}
-	s="$1/$s"
-	for k in $(find $s -name '*.tar.xz'); do
-		btf=$(basename ${k%%.tar.xz})
-		name="$distro-${btf%%.btf}"
-		name=$(echo $name | tr \\055\\056\\057 _)
-		tar xf $k || die "oh noes"
-		if ./quark-btf -g $name -f $btf 2>/dev/null; then
-			echo "$name OK" 1>&2
-			Good+=("$name")
-		else
-			echo "$name FAIL" 1>&2
-		fi
-		rm -f $btf
-		Total=$((Total+1))
+	if [ $1 = amd64 ]; then
+		Arch=amd64
+		Srcs="$Srcs_amd64"
+	elif [ $1 = arm64 ]; then
+		Arch=aarch64
+		Srcs="$Srcs_arm64"
+	else
+		die "bad arch"
+	fi
+
+	printf "#ifdef __%s__\n\n" $Arch
+
+	for s in $Srcs; do
+		distro=${s%/*}
+		s="$Btfhub_path/$s"
+		for k in $(find $s -name '*.tar.xz'); do
+			btf=$(basename ${k%%.tar.xz})
+			name="$distro-${btf%%.btf}"
+			name=$(echo $name | tr \\055\\056\\057 _)
+			tar xf $k || die "oh noes"
+			if ./quark-btf -g $name -f $btf 2>/dev/null; then
+				echo "$name OK" 1>&2
+				if [ $1 = amd64 ]; then
+					Good_amd64+=("$name")
+				elif [ $1 = arm64 ]; then
+					Good_arm64+=("$name")
+				fi
+			else
+				echo "$name FAIL" 1>&2
+			fi
+			rm -f $btf
+			Total=$((Total+1))
+		done
 	done
-done
+
+	printf "#endif /* __%s__ */\n\n" $Arch
+
+}
+
+do_arch amd64
+do_arch arm64
 
 cat <<EOF
 struct quark_btf *all_btfs[] = {
 EOF
 
-for name in "${Good[@]}"; do
-	printf "\t&%s,\n" "$name"
-done
+	printf "\n#ifdef __amd64__\n"
```

Comment:
Weird spacing in these 10 lines.

Reply:
whoa, good catch, no clue how I didn't spot it.

Reply:
arghh ah no, wait, sorry, lol.
Now I remembered, this is intentional, it's to look like it's inside the "all_btfs" block.
Does it make sense?

Reply:
Oh, I see. I wouldn't have figured that out, since the printf's still need to have `\t`s in them.